### ISSUE TITLE
Compilation fails on Ubuntu Jammy aarch64

### ISSUE THREAD

**Issue by alfa07** [oxidecomputer/usdt#66]
On master the compilation fails with:
```
error: invalid register `al`: unknown register
   --> usdt-impl/src/record.rs:116:17
    |
116 |                 inout("al") version,
    |                 ^^^^^^^^^^^^^^^^^^^

error: could not compile `usdt-impl` due to previous error

```

```
ubuntu@u-22-04-jammy-main:~/usdt$ uname -a
Linux u-22-04-jammy-main 5.15.0-52-generic #58-Ubuntu SMP Thu Oct 13 08:04:57 UTC 2022 aarch64 aarch64 aarch64 GNU/Linux
```
```
ubuntu@u-22-04-jammy-main:~/usdt$ git log
commit eac0fe5f03c3fbf23468ead5cb140f62d51ac3f3 (HEAD -> master, origin/master, origin/HEAD)
Author: Patrick Mooney <pmooney@pfmooney.com>
Date:   Wed Apr 20 20:07:46 2022 -0500

    Handle ASM features more gracefully

```
```
ubuntu@u-22-04-jammy-main:~/usdt$ rustc --version
rustc 1.65.0 (897e37553 2022-11-02)
```

**Comment by bnaecker** [comment]
Thanks for reporting this @alfa07. Looks like you're building on an ARM, which is only really supported on macOS at the moment. I have a fix from earlier in the year, that I seem to have forgotten to finalize and push, which extends support by naming the right registers on other ARM platforms. I'll take a look at that soon, and try to get this sorted.

So I understand better, are you building this crate directly, or as part of a dependency of something else? We've only done minimal testing on Linux platforms, given that DTrace is not very well-supported. (If at all, I'm not sure of the state.) To be clear, you should definitely be able to build this crate and inspect the resulting binaries, but I just want to make sure you are aware of the limitations on Linux!

**Comment by alfa07** [comment]
@bnaecker thank you for looking into it, yes, I am building on macOS 12.6 (Chip: Apple M1 Max) laptop inside ubuntu container :)

**Comment by bnaecker** [comment]
Closed by #79. @alfa07 Let me know if doesn't resolve your problems. I don't have an ARM system available for testing, but the offending lines of code are no longer arch-specific so it _should_ be resolved. Thanks!
